device/tt_device.h

```diff
@@ -623,6 +623,20 @@ class tt_SiliconDevice: public tt_device
                     const uint32_t &num_host_mem_ch_per_mmio_device = 1, const bool skip_driver_allocs = false,
```

Comment:
I'd like us to remove completely ndesc_path from both constructors.
It will be useful to leave the option for custom sdesc_path, and for custom set of target_devices (we can define later how exactly would this be used, probably need an api_test for this). But I'd nuke cluster_desc yaml completely from the interface.

Reply:
I want to make this change transparent for all clients at the moment, meaning they can still rely on old constructor. I can change this in a separate PR. We would just want to confirm that there is no value in passing custom cluster descriptor, intuitively it looks that it doesn't have any real value or use case